lib/pos_remote_node_indexer.go

```diff
@@ -0,0 +1,231 @@
+package lib
+
+import "github.com/deso-protocol/core/bls"
+
+// RNIndexId is a custom string type used for identifying different types of remote node indices.
+type RNIndexId string
+
+// Constants for different types of remote node indices.
+const (
+	RNIndexId_Validator              RNIndexId = "VALIDATOR"
+	RNIndexId_ValidatorAttempted     RNIndexId = "VALIDATOR_ATTEMPTED"
+	RNIndexId_NonValidator_Outbound  RNIndexId = "NONVALIDATOR_OUTBOUND"
+	RNIndexId_NonValidator_Inbound   RNIndexId = "NONVALIDATOR_INBOUND"
+	RNIndexId_NonValidator_Attempted RNIndexId = "NONVALIDATOR_ATTEMPTED"
+)
```

Comment:
IMO, there's no need for these identifiers. When an external caller calls the `GetValidatorIndex()` function, they know that the index returned will be the validator index. They don't additionally need a flag inside the index that identifies its type.

Removing this will make your code and the maps indexer more concise